### PR TITLE
Add some more details to testing guide

### PR DESCRIPTION
**Pull request by kmahar**

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -1,10 +1,8 @@
 # Testing 
 
-Running `swift test` from the terminal will trigger the unit tests (or triggering the test action in Xcode).
+SwiftPM is integrated with [XCTest, Apple’s unit test framework](https://developer.apple.com/documentation/xctest). Running `swift test` from the terminal, or triggering the test action in Xcode, will run all of your XCTest test cases. Test results will be displayed in Xcode or printed out to the terminal.
```
Comment (ktoso) [inline docs/testing.md:3]: Maybe we can be a bit less xcode specific and just say "IDE (like Xcode or similar)" and then continue using IDE?

Comment (ktoso) [reply docs/testing.md:3]: Love the intro tho, it is super unclear to newcomers what the hell xctest is and if they have to use it or not :)

**Review by ktoso — APPROVED** [review]
LGTM, minor nitpick

File: docs/testing.md (hunk 1 above)
Comment (kmahar) [reply docs/testing.md]: good point, done ✅

**Review by tomerd — APPROVED** [review]

**Pull request merged**